src/world.rs

```diff
@@ -420,22 +409,29 @@ impl<C> LazyUpdateInternal for LazyRemovalInfo<C>
     where C: Component + Send + Sync
 {
     fn update(self: Box<Self>, world: &World) {
-        world.write::<C>().remove(self.entity);
+        world.write::<C>().remove(self.0);
     }
 }
 
-/// Lazy updates can be used after creating a new
-/// entity in a system. This way, none of the actual
-/// component storages have to be borrowed mutably.
-///
+impl LazyUpdateInternal for Box<Fn(&World) + Send + Sync> {
```

Comment:
Did you try `impl<F> LazyUpdateInternal for F where F: Fn(&World) + Send + Sync + 'static`?